src/uffd_handler/src/main.rs

```diff
@@ -53,6 +52,17 @@ pub enum Error {
 
 type Result<T> = std::result::Result<T, Error>;
 
+fn validate_snapshot_file(file_name: &String) -> Result<(File, usize)> {
```

Comment:
Mh, I think something weird is going on with clippy. There's a lint about using `&str` instead of `&String` (`clippy::ptr_arg`). If I copy this function from `main.rs` to `lib.rs` then the lint fires 🤔.

Reply:
Hmm, I experienced the same behavior. I will take a look into why clippy doesn't catch all errors. In the meantime, I fixed the code.